src/style.js

```diff
@@ -5,15 +5,18 @@ exports.register = function(linter) {
   // deprecated and then re-introduced for ES6.
 
   linter.on("Identifier", function style_scanProto(data) {
-    if (linter.getOption("proto")) {
+    var esnext = linter.getOption("esnext");
+
+    if (linter.getOption("proto") || (esnext && (linter.getOption("browser") ||
+        linter.getOption("node") || linter.getOption("rhino")))) {
```

Comment:
I spent a while looking at this line thinking it was wrong, till noticing the `(esnext && ...` - any chance we can reformat this to make it clearer?

e.g.
```
    if (linter.getOption("proto") ||
       (esnext && (linter.getOption("browser") || linter.getOption("node") || linter.getOption("rhino")))) {
```

So.. is `__proto__` annex b ? and node and rhino are classed as "browser environments" ?

If that's the case I wonder whether the error message should be more specific? if you didn't have the node option on, it wouldn't be obvious that the node option switches it off ?